server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -204,13 +204,18 @@ private SearchHit createSearchHit(SearchContext context,
                                       Map<String, Set<String>> storedToRequestedFields,
                                       LeafReaderContext subReaderContext) {
         if (fieldsVisitor == null) {
-            return new SearchHit(docId, null, null);
+            return new SearchHit(docId, null, null, null);
+
         }
 
         Map<String, DocumentField> searchFields = getSearchFields(context, fieldsVisitor, subDocId,
             storedToRequestedFields, subReaderContext);
 
-        SearchHit searchHit = new SearchHit(docId, fieldsVisitor.id(), searchFields);
+        Map<String, DocumentField> metaFields = new HashMap<>();
+        Map<String, DocumentField> documentFields = new HashMap<>();
+        splitSearchHitFields(searchFields, documentFields, metaFields);
```

Comment:
Can we use `SearchHit::splitFieldsByMetadata` instead of introducing a new function?

Reply:
done